plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-webhook-handler.php

```diff
@@ -68,9 +74,11 @@ private function process_checkout_order_approved( $event ) {
 				)
 			);
 
-			// Capture the payment after approval.
+			// Authorize or capture the payment after approval.
 			if ( 'CAPTURE' === $event['resource']['intent'] ) {
-				$this->capture_payment( $order, $event['resource']['links'] );
+				$this->authorize_or_capture_payment( $order, $event['resource']['links'], 'capture' );
+			} else {
+				$this->authorize_or_capture_payment( $order, $event['resource']['links'], 'authorize' );
```

Comment:
Nit: Might be easier to read if we perform the if-else/ternary logic on the action string.